.github/workflows/go.yml

```diff
@@ -0,0 +1,99 @@
+# Workflow will:
+# - install Golang dependencies;
+# - lint and run tests with multiple Golang versions.
+# For more information see:
+# https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-go
+
+name: test_application
+
+on:
+  workflow_call:
+    inputs:
+      app_name:
+        type: string
+        required: true
+        description: Application and source code folder name
+      ci_image:
+        type: string
+        required: false
+        default: ubuntu-22.04
+        description: Github type of machine to run the workflow on
+      go_versions:
+        type: string
+        required: false
+        default: "['1.21','1.22','1.23']"
+        description: Golang versions to build workflow matrix
```

Comment:
Why do you want to build so many versions? You have limited CI minutes, just build the latest one unless you have a good reason